### DISCUSSION TITLE
Smooth Scroll

### DISCUSSION THREAD

**Discussion by ilyadegroeve** [sumatrapdfreader/sumatrapdf#4122] [Ideas]
Adding the feature "Smooth Scroll". Currently scrolling in Sumatra PDF is low fps and very scratchy, smooth scrolling would solve this issue.

**Comment by GitHubRulesOK** [comment]
Scroll behaviour is 2 parts so hardware and software have a different effect for each user. Personally I use smooth auto scroll that is built into my mouse, (same one I use in browser to make that smooth) but each mouse may have different "smoothness" Note it is reasonably smooth enough even though SumatraPDF advanced settings are currently  `SmoothScroll = false`

The software limit is that the rendering needs to be built lines at a time so often a page may stutter as each text high line is rendered since PDF are not like an image format which can show scan line by line.

Comparatively in my hardware Acrobat and SumatraPDF are equally smooth.

**Comment by ilyadegroeve** [comment]


**Reply by GitHubRulesOK** [reply]
FireFox PDF reader ? if you mean Mozilla PDFasAnIMG.js then that is not a normal binary (Use Adobe Reader as a Compliant yardstick) 
If you inspect the Firefox page is is scanlines I.E it is a picture of a PDF just like GoogleViewer or FlipBooks are NOT a PDF so images will always scroll by exceptionally small steps compared to vector based rendering. 

And even this knackered $1 mouse autoscroll in SumatraPDF is as smooth as here in Edge. The Powered by Adobe viewer in edge looks jerkier to me than using SumatraPDF in old Firefox or other newer current browsers.

Acrobat Reader DC in FireFox variants is way smoother than Powered by Acrobat in Chromium Edge And if I switch Reader DC on in Edge it is amazingly fast (as its lighter that standalone!) even looks as fast as SumatraPDF

**Reply by lokmac** [reply]
> I am talking about auto scroll (middle mouse button).

You should update your original post because that is not called "smooth scroll" (which would be smooth animation of scrolling when using the wheel).

**Reply by lokmac** [reply]
But I think there is room for improvement of the threshold for auto-scroll using the middle mouse button. The threshold of the speed relative to distance from click should be less. I can see that if I click maybe 1/3 down the screen, scoll down is significantly faster than scroll up. 

I think full speed should be 1/3 of the screen away from centre to cater for the ability to achieve full speed auto-scroll in both directions for the majority of clicks (OR make it relative, so it reaches full speed by the time the cursor is moved to the maximum point of the screen above or below the click, but I think that might be confusing and prevent muscle memory movements as the scale will be different every time - a fixed scale is better i think).

**Reply by ilyadegroeve** [reply]
I don't know why I switched it to auto scroll, I still have the same issue with normal scroll and auto scroll. 
In this video you can see my comparison

https://github.com/user-attachments/assets/f499ca01-6c2f-4f45-8ee7-d4d81fb7ef09

**Reply by lokmac** [reply]
Ahh ok. In which case kjk's response is the answer after all.

**Comment by kjk** [comment]
Sadly, smooth scroll is not easy to implement. Web browsers have very sophisticated rendering engine that required lots of people and lots of time to implement. I'm one person and I don't see an easy way to improve scrolling.

**Comment by Av1ators** [comment]
I will wait :D

**Comment by Calvorejas** [comment]
Hi,

Sorry for "reopening" an old topic here but I wanted to comment on this as well. As hard as it may be to implement this feature... I have to ask for this to be implemented... I can use the program as is... I would really prefer it to have smooth scrolling but whatever...

The problem is that, as good as the program is... I just got feedback from a friend I recommended this software to... And the first thing he told me was:

"The scroll sucks man, it works at like 3 fps or something, I hate it. I don't want to go back to Adobe but man, I don't like this either".

As good as the program is (because I know it's good)... If the first impressions are bad... New people will discard this software and search for a "nicer" experience elsewhere...

Today I had to scroll on a document that we use to study in class... I had to be precise, and I just could not... I ended up having to do it with the scroll bar arrows because it was just impossible to scroll so slightly just enough to get to where you want.

Hope this gets reconsidered at any moment because it would really help for keeping newcomers, as well as helping actual users of this program who just want to have a better experience. This is less noticeable with a regular mouse, but when you use a TouchPad... Very bad. Again, hope there is something to be done here.

Kind regards.

**Reply by kjk** [reply]
Smooth scroll is implemented in 3.7 [pre-release](https://www.sumatrapdfreader.org/prerelease)

See: https://www.sumatrapdfreader.org/docs/Version-history#:~:text=SmoothScroll

Make sure `SmoothScroll` is set to true: https://www.sumatrapdfreader.org/docs/Advanced-options-settings#:~:text=SmoothScroll%20%3D%20true

**Comment by vu205** [comment]
Is there a way when I hold shift then speed of scroll increase x2, x3 ( we can change velocity of speed ) ?
